admin.go

```diff
@@ -894,16 +897,30 @@ func (h adminHandler) originAllowed(origin *url.URL) bool {
 	return false
 }
 
+// etagHasher returns a the hasher we used on the config to both
+// produce and verify ETags.
+func etagHasher() hash.Hash32 { return fnv.New32a() }
+
 func handleConfig(w http.ResponseWriter, r *http.Request) error {
 	switch r.Method {
 	case http.MethodGet:
 		w.Header().Set("Content-Type", "application/json")
-
-		err := readConfig(r.URL.Path, w)
+		// Set the ETag as a trailer header.
+		// The alternative is to write the config to a buffer, and
+		// then hash that.
+		w.Header().Set("Trailer", "ETag")
```

Comment:
Though unconventional, this is allowed: https://httpwg.org/specs/rfc9110.html#rfc.section.8.8.3

Yay for RFC 9110 -- HTTP Semantics -- which says:

> A sender MAY send the ETag field in a trailer section (see [Section 6.5](https://httpwg.org/specs/rfc9110.html#trailer.fields)). However, since trailers are often ignored, it is preferable to send ETag as a header field unless the entity tag is generated while sending the content.

So, I dunno. I'd say let's go ahead and try this as a trailer and if we get complaints with _valid technical reasons_ why trailer won't work for this, then maybe we can switch to header + buffering.